src/logging.cpp

```diff
@@ -165,10 +187,16 @@ const CLogCategoryDesc LogCategories[] =
 #endif
     {BCLog::UTIL, "util"},
     {BCLog::BLOCKSTORE, "blockstorage"},
-    {BCLog::ALL, "1"},
     {BCLog::ALL, "all"},
+    {BCLog::NONE, "0"},
+    {BCLog::ALL, "1"},
 };
 
+// Ignore last 2 extra mappings of NONE and ALL.
+const std::unordered_map<BCLog::LogFlags, std::string> LogCategoryToStr{
+    LogCategories,
+    LogCategories + std::size(LogCategories) - 2};
+
```

Comment:
In the first commit, a few suggestions
```diff
@@ -138,7 +138,7 @@ struct CLogCategoryDesc {
     }
 };
 
-const CLogCategoryDesc LogCategories[]{
+static CLogCategoryDesc LogCategories[]{
     {BCLog::NONE, "none"},
     {BCLog::NET, "net"},
     {BCLog::TOR, "tor"},
@@ -174,10 +174,10 @@ const CLogCategoryDesc LogCategories[]{
     {BCLog::ALL, "1"},
 };
 
-// Ignore last 2 extra mappings of NONE and ALL.
-const std::unordered_map<BCLog::LogFlags, std::string> LogCategoryToStr{
+static std::unordered_map<BCLog::LogFlags, std::string> LogCategoryToStr{
     LogCategories,
-    LogCategories + std::size(LogCategories) - 2};
+    LogCategories + std::size(LogCategories) - 2 // ignore last 2 extra mappings of NONE and ALL
+};
```

Reply:
Wouldn't it make sense to do it the other way around?

First build `LogCategoryToStr`, then build `LogCategoryDesc` from that, adding the ALL=1 and NONE=0 instead of cutting them off. I think that's neater and less error prone.

Reply:
> In the first commit, a few suggestions

Done 👍🏻 

> First build LogCategoryToStr, then build LogCategoryDesc from that, adding the ALL=1 and NONE=0 instead of cutting them off. I think that's neater and less error prone.

Agreed, building `LogCategories` from `LogCategoryToStr` is definitely cleaner!

However, I am unable to think of a trivial method to implement it as currently both `LogCategories` and `LogCategoryToStr` are global variables where the values are added in its constructor. To build `LogCategories` from `LogCategoryToStr`, we will most probably need to change `LogCategories` type from an array to a class that allows us to add our own construction logic (build from `LogCategoryToStr` and add `ALL=1` and `NONE=0`).

If changing `LogCategories` to a class is the desired approach, I will split the first commit into a separate PR.